libs/api/domains/work-machines/src/lib/workMachines.resolver.ts

```diff
@@ -181,4 +182,19 @@ export class WorkMachinesResolver {
       subCategory,
     )
   }
+
+  @Scopes(ApiScope.vinnueftirlitid)
+  @Query(() => MachineType)
+  @Audit()
+  async getTypeByRegistrationNumber(
+    @CurrentUser() auth: User,
+    @Args('registrationNumber') registrationNumber: string,
+    @Args('applicationId') applicationId: string,
+  ) {
+    return this.workMachinesService.getTypeByRegistrationNumber(
+      auth,
+      registrationNumber,
+      applicationId,
+    )
+  }
```

Comment:
Does this need to be a new query? The `machineType` looks more like a property of a `WorkMachine` (on the outside at least) so why not add it to the `WorkMachine` model instead of creating a whole new thing? 
Then, you can use a field resolver to fetch the `machineType` if queried for

Reply:
Is this new type (returned by `apiMachineTypesTypeByRegistrationNumberGet`) different from the type returned by the `getWorkMachineById`?

Reply:
Approve for now, would be nice to review the domain later and maybe draft a v2